optuna/pruners/hyperband.py

```diff
@@ -70,20 +77,36 @@ class HyperbandPruner(BasePruner):
     """
 
     def __init__(
-        self, min_resource=1, reduction_factor=3, n_brackets=4, min_early_stopping_rate_low=0
-    ):
-        # type: (int, int, int, int) -> None
-
+        self,
+        max_resource: int,
+        min_resource: int = 1,
+        reduction_factor: int = 3,
+        n_brackets: Optional[int] = None,
+        min_early_stopping_rate_low: int = 0,
+    ) -> None:
+        self._max_resource = max_resource
         self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._reduction_factor = reduction_factor
         self._resource_budget = 0
-        self._n_brackets = n_brackets
+
+        if n_brackets is None:
+            self._n_brackets = numpy.ceil(numpy.log2(max_resource) / numpy.log2(reduction_factor))
+            self._n_brackets = int(self._n_brackets)
+        else:
+            message = (
+                "The use of `HyperbandPruner.n/brackets` is deprecated. "
```

Comment:
"The use of `HyperbandPruner.n_brackets` is deprecated" sounds like there is an attribute named `n_brackets` and it's now deprecated to me.

```suggestion
                "The argument of `n_bracket` is deprecated. "
```